src/app/plugins/give.plugin.ts

```diff
@@ -0,0 +1,67 @@
+import { MessageEmbed } from 'discord.js';
+import Constants from '../../common/constants';
+import { Plugin } from '../../common/plugin';
+import { IContainer, IMessage, ChannelType } from '../../common/types';
+import { PointsDocument } from '../../schemas/points.schema';
+
+export default class GivePlugin extends Plugin {
+  public commandName: string = 'give';
+  public name: string = 'Give Plugin';
+  public description: string = 'Give your tacos to someone else';
+  public usage: string = 'give @Tanndlin all\n give @Tanndlin 100';
+  public override pluginAlias = [];
+  public permission: ChannelType = ChannelType.Public;
+  public override pluginChannelName: string = Constants.Channels.Public.Games;
+
+  public override commandPattern: RegExp = /(all|\d+)/;
+
+  constructor(public container: IContainer) {
+    super();
+  }
+
+  public override validate(message: IMessage, args: string[]): boolean {
+    if (args.length !== 2) {
+      return false;
+    }
+
+    return message.mentions.users.size === 1 && this.commandPattern.test(args[1]);
+  }
+
+  public async execute(message: IMessage, args: string[]) {
+    const userDoc = await this.container.pointService.getUserPointDoc(message.author.id);
+    const recipientUser = message.mentions.users.first()!; // Asserting true after validate
+    const recipient = await this.container.pointService.getUserPointDoc(recipientUser.id);
```

Comment:
```suggestion
    const recipientUser = message.mentions.users.first()!; // Asserting true after validate
    const [userDoc, recipient] = await Promise.all([message.author.id, recipientUser.id].map(id => this.container.pointService.getUserPointDoc(id)));
```
These async tasks can run concurrently.